lib/value/src/kind/comparison.rs

```diff
@@ -189,6 +216,7 @@ impl Kind {
             || self.is_timestamp()
             || self.is_regex()
             || self.is_null()
+            || self.is_undefined()
```

Comment:
This is another one in which its outcome is muddied by the current API.

If I have a value, that can either exist and be a string, or not exist at all, I consider the type of that value to be "exact" _when the value exists_. When the value doesn't exist, there's no concept of "exactness" of the type, since there's no type to tie to a non-existing value.

Reply:
> When the value doesn't exist, there's no concept of "exactness" of the type
It is very cleanly defined as "at most one type is set". `undefined` is a normal type, similar to `null`.